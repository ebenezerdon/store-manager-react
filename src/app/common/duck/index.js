export {
  doFetchCurrentUser,
  doFetchAllUsers,
  doFetchSaleRecord,
  doFetchProducts
} from './operations';
export { default as fetchAllDataReducer } from './reducers';
export { default as constants } from './constants';
export { default as types } from './types';
export { default as actions } from './actions';
