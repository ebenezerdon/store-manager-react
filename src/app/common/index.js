export { fetchAllDataReducer } from './duck';
export { constants } from './duck';
export { EllipsisLoaderComponent } from './loaders';

export {
  doFetchCurrentUser,
  doFetchAllUsers,
  doFetchSaleRecord,
  doFetchProducts
} from './duck';
