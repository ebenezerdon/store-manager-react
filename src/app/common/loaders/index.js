export { default as EllipsisLoaderComponent } from './EllipsisLoaderComponent';
