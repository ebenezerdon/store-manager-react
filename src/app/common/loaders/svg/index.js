export { default as ellipsisLoader } from './EllipsisLoader.svg';