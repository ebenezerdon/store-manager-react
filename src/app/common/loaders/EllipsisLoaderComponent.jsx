import React from 'react';
import { ellipsisLoader } from './svg';

const EllipsisLoaderComponent = () => {
  return <img className="ellipsis" src={ellipsisLoader} alt="loader" />;
};

export default EllipsisLoaderComponent;
