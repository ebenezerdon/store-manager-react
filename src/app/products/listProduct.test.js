import React from 'react';
import { shallow, mount } from 'enzyme';
import { EllipsisLoaderComponent } from '../common/loaders'
import ListProductComponent from './ListProductsComponent';
import {
  mapDispatchToProps,
  mapStateToProps,
} from './ListProductsContainer';

describe('ListProduct Container', () => {
  it('should show initial state', () => {
    const initialState = {
      fetchAllData: { cart: { Data: ['Data'] } },
      errorMessage: '',
      checkout: {
        checkoutState: '',
      },
    };
    expect(mapStateToProps(initialState).cart).toEqual({ Data: [ 'Data' ] });
  });

  it('should dispatch action', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).fetchStoreData();
    mapDispatchToProps(dispatch).addToCart();
    expect(typeof dispatch.mock.calls[0][0]).toEqual('function');
  });
});

describe('List products component', () => {
  it('should render the component', () => {
    const fetchStoreData = jest.fn();
    const products = {};
    const component = shallow(
      <ListProductComponent
        {...{ fetchStoreData, products }}
      />,
    );
    expect(component.exists()).toBe(true);
    expect(component).toMatchSnapshot();
    expect(fetchStoreData).toHaveBeenCalled();
  });
});
