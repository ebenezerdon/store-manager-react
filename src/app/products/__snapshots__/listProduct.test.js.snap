// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List products component should render the component 1`] = `
<ListProductComponent
  fetchStoreData={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  products={Object {}}
/>
`;
